packages/flutter/lib/src/material/menu_bar.dart

```diff
@@ -0,0 +1,3256 @@
+      return false;
+    }
+    if (focusedItem.parent == controller.root) {
+      if (controller.openMenu == null) {
+        controller.openMenu = focusedItem;
+        return true;
+      }
+      final List<_MenuNode> children = focusedItem.focusableChildren;
+      if (children.isNotEmpty) {
+        controller.openMenu = children[0];
+      }
+      return true;
+    }
+    _MenuNode? nextFocusable = focusedItem.nextSibling;
+    while (nextFocusable != null && !nextFocusable.focusNode!.canRequestFocus) {
+      nextFocusable = nextFocusable.nextSibling;
+    }
+    if (nextFocusable != null) {
+      controller.openMenu = nextFocusable;
+    }
+    return true;
+  }
+
+  @override
+  void invoke(DirectionalFocusIntent intent) {
+    final TextDirection textDirection = Directionality.of(controller.menuBarContext);
+    switch (intent.direction) {
+      case TraversalDirection.up:
+        if (_moveUp()) {
+          return;
+        }
+        break;
+      case TraversalDirection.down:
+        if (_moveDown()) {
+          return;
+        }
+        break;
+      case TraversalDirection.left:
+        switch (textDirection) {
+          case TextDirection.rtl:
+            if (_moveForward()) {
+              return;
+            }
+            break;
+          case TextDirection.ltr:
+            if (_moveBackward()) {
+              return;
+            }
+            break;
+        }
+        break;
+      case TraversalDirection.right:
+        switch (textDirection) {
+          case TextDirection.rtl:
+            if (_moveBackward()) {
+              return;
+            }
+            break;
+          case TextDirection.ltr:
+            if (_moveForward()) {
+              return;
+            }
+            break;
+        }
+
+        break;
+    }
+    super.invoke(intent);
+  }
+}
+
+/// Provides default implementation for [MenuItem] in a [StatefulWidget], to serve
+/// as a base class for the [MenuBarItem], [MenuItemGroup] and [MenuBarMenu] classes.
+abstract class _MenuBarItemDefaults extends StatefulWidget implements PlatformMenuItem {
+  const _MenuBarItemDefaults({
+    super.key,
+    required this.label,
+    this.labelWidget,
+  });
+
+  /// A required label displayed on the entry for this item in the menu.
+  ///
+  /// This is rendered by default in a [Text] widget.
+  /// The label appearance can be overridden by using a [labelWidget] to render
+  /// a different widget in its place.
+  ///
+  /// This label is also used as the default [semanticLabel].
+  @override
+  final String label;
+
+  /// An optional widget that will be displayed in place of the default [Text]
+  /// widget containing the [label].
+  final Widget? labelWidget;
+
+  @override
+  MenuSerializableShortcut? get shortcut => null;
+
+  /// The function called when the mouse leaves or enters this menu item's
+  /// button.
+  ValueChanged<bool>? get onHover;
+
+  /// The list of top-level menu items to show in the [MenuBar].
```

Comment:
This seems to be different than what the base class indicated.

Reply:
Removed the comment.